### PR TITLE
test: local server for mocking tests

### LINKED ISSUE #161: We need a local server for mocking our tests.
This issue is related to https://github.com/asyncapi/cli/pull/110#discussion_r768405925

### PR DESCRIPTION
**Pull request by chinma-yyy**
**Description**

- Added mock server implementation which spins up a local server for testing `parseFromUrl`
- Modified one test for testing it 
- Added the remote files to `dummyspec` directory

**Related issue(s)**

Resolves #161

### CODE REVIEW

File: test/commands/generate/models.test.ts [hunk 1]
```diff
@@ -2,14 +2,22 @@
...
 import path from 'path';
 import { test } from '@oclif/test';
+import { createMockServer, stopMockServer } from '../../testHelper';
 const generalOptions = ['generate:models'];
 const outputDir = './test/commands/generate/models';
 
 describe('models', () => {
+
+  beforeAll(()=>{
+    createMockServer();
```
Comment (chinma-yyy) [inline test/commands/generate/models.test.ts:12]: I was thinking that should i add some variable here for the port we are using so we can use it in the URLs .

**Comment by chinma-yyy** [comment]
Hey @derberg and @Souvikns , I have implemented the basic server implementation. So if you can review this then I can move ahead to make changes to the tests which use remote files and add the files to the dummyspec directory. I think to make the implementation more reusable and flexible.

**Comment by chinma-yyy** [comment]
@Souvikns

**Comment by chinma-yyy** [comment]
@derberg should i go ahead and make changes to all other tests?

File: test/testHelper.ts [hunk 2]
```diff
@@ -80,3 +85,45 @@ export default class ContextTestingHelper {
 export function fileCleanup(filepath: string) {
   unlinkSync(filepath);
 }
+
+export function createMockServer (port = 8080) {
+  server = http.createServer((req,res) => {
+    if (req.method ==='GET') {
+      const filePath= path.join(SERVER_DIRECTORY, req.url || '/');
+      fs.readFile(filePath, (error, content) => {
```
[the remark this answers is not in the document]
Comment (chinma-yyy) [reply test/testHelper.ts:93]: Okay:v:

File: test/commands/generate/models.test.ts (hunk 1 above)
Comment (derberg) [reply test/commands/generate/models.test.ts:11]: not sure what you mean, can you explain?

File: test/dummyspec/streetlights.yml [hunk 3]
```diff
@@ -0,0 +1,209 @@
+asyncapi: '2.0.0'
```
Comment (derberg) [inline test/dummyspec/streetlights.yml:1]: as this is only for the tests related to fetching schema from URL, let's make it super super dummy minimalistic file, so please rename to `dummyFetchTest.yml` and content like
```
asyncapi: 2.6.0
info:
  title: Dummy example used in tests related to fetching AsyncAPI from URL
  version: 0.0.1
channels: {}
```

**Review by derberg — CHANGES_REQUESTED** [review]
yeah, use it everywhere `https://bit.ly/asyncapi` is used

File: test/dummyspec/streetlights.yml (hunk 3 above)
Comment (derberg) [reply test/dummyspec/streetlights.yml]: @chinma-yyy yo, I think you forgot about this one

Comment (chinma-yyy) [reply test/dummyspec/streetlights.yml]: No, actually the `convert` command tests depend on this older version spec. So,if the version is changed the test suite will fail :v:

Comment (derberg) [reply test/dummyspec/streetlights.yml]: then change to `2.5`. More important is to have dummy, non streetlight example

**Comment by chinma-yyy** [comment]
I think this PR is ready @derberg @Souvikns . Let me know for any more changes

**Review by Souvikns — APPROVED** [review]
Looks good to me 👍🏼 , wait for @derberg's review before merging

**Review by derberg — APPROVED** [review]
Thanks a lot!!!

**Comment by derberg** [comment]
@allcontributors please add @chinma-yyy for code, test

**Pull request merged**